block/block_sync.go

```diff
@@ -99,6 +99,9 @@ func (bSyncService *BlockSyncService) initBlockStoreAndStartSyncer(ctx context.C
 // Note: Only returns an error in case block store can't be initialized. Logs
 // error if there's one while broadcasting.
 func (bSyncService *BlockSyncService) WriteToBlockStoreAndBroadcast(ctx context.Context, block *types.Block) error {
+	if bSyncService.genesis.InitialHeight < 0 {
+		return fmt.Errorf("invalid initial height; cannot be negative")
+	}
```

Comment:
Since `InitialHeight` is a `uint64`, this is something that's never possible right?

Reply:
`genesis.InitialHeight` is of type `int64` so it might overflow.

However, we're already validating this in `GenesisDoc.ValidateAndComplete` so this check is redundant. We have `ValidateAndComplete` checks whenever a new genesis doc is created, however we can probably check this again when reading the doc in calls like `GenesisDocFromFile`.